HyperVExtension/src/HyperVExtension/CommunicationWithGuest/WmiUtility.cs

```diff
@@ -124,4 +138,47 @@ public static bool JobCompleted(ManagementBaseObject outParams, ManagementScope
 
         return jobCompleted;
     }
+
+    public static FeatureAvailabilityKind GetHyperVFeatureAvailability()
+    {
+        try
+        {
+            var searcher = new ManagementObjectSearcher($"SELECT * FROM Win32_OptionalFeature WHERE Name = 'Microsoft-Hyper-V'");

```

Comment:
NIT: Declare the columns instead of using *.  This protects against
1. Returning more data than needed
2. Columns changing location, or new columns being added.

Also, adding the column names is more declarative of your intentions.